WordPress/Classes/Stores/StatsPeriodStore.swift

```diff
@@ -1473,3 +1489,208 @@ private extension StatsPeriodStore {
         })
     }
 }
+
+// MARK: - Data for Views & Visitors weekly details
+/// We require to fetch summary for .day period and referrers and countries for .week period for insights
+/// Since StatsPeriodStore is designed to function with one stats period at one time, we require separate methods
+/// that explicitly set "week" or "day" period when fetching particular data
+
+private extension StatsPeriodStore {
+    func shouldFetchViewsAndVisitors() -> Bool {
+        return [state.summaryStatus,
+                state.topReferrersStatus,
+                state.topCountriesStatus].first { $0 == .loading } == nil
+    }
+
+    func setViewsAndVisitorsFetchingStatus(_ status: StoreFetchingStatus) {
+        if FeatureFlag.statsPerformanceImprovements.enabled {
```

Comment:
Maybe we don't need the statsPerformanceImprovements flag anymore? The feature flag is enabled + production has been running with this enabled for awhile

Reply:
Yes, it does look like the flag could be removed.